arangod/Pregel/Algos/ReadWrite.cpp

```diff
@@ -0,0 +1,130 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Roman Rabinovich
+////////////////////////////////////////////////////////////////////////////////
+
+#include "ReadWrite.h"
+
+#include <utility>
+#include "Pregel/Aggregator.h"
+#include "Pregel/GraphFormat.h"
+#include "Pregel/Iterators.h"
+#include "Pregel/MasterContext.h"
+#include "Pregel/Utils.h"
+#include "Pregel/VertexComputation.h"
+
+using namespace arangodb;
+using namespace arangodb::pregel;
+using namespace arangodb::pregel::algos;
+
+static std::string const simulatedAggregatorName = "simulatedAggregator";
+
+struct ReadWriteWorkerContext : public WorkerContext {
+  ReadWriteWorkerContext() = default;
+};
+
+ReadWrite::ReadWrite(application_features::ApplicationServer& server,
+                     VPackSlice const& userParams)
+    : SimpleAlgorithm(server, "readwrite", userParams) {}
+
+struct ReadWriteGraphFormat final : public GraphFormat<V, E> {
+  ReadWriteGraphFormat(application_features::ApplicationServer& server,
+                       std::string sourceFieldName, std::string resultFieldName)
+      : GraphFormat(server),
+        sourceFieldName{std::move(sourceFieldName)},
+        resultFieldName{std::move(resultFieldName)} {}
+  std::string const sourceFieldName;
+  std::string const resultFieldName;
+
+  void copyVertexData(arangodb::velocypack::Options const&,
+                      std::string const& documentId, VPackSlice document,
+                      V& targetPtr, uint64_t& /*vertexIdRange*/) override {
+    if (!document.hasKey(sourceFieldName)) {
```

Comment:
@romanatarango 
it's better to avoid `hasKey` + `get` pattern as key lookup happens twice in this case.
Better to use
```
auto const val = document.get(sourceFieldName);

if (val.isNone()) {
   // handle "no property" case
}
...
```

Reply:
Done